### PR TITLE
ParticipantConfig: add Include Semantics support

### PR DESCRIPTION
**Pull request by VJanKraemer**
## Subject

* Add support to include config snippets in configs provided via file
* Add support to include config snippets in configs provided as string
* Move file uitility functions to their own unit
* Add tests for all provided funtionality

## Description
Issues: SILKIT-1391, SILKIT-1495

## Developer checklist (address before review)

- [x] Changelog.md updated
- [ ] Prepared update for depending repositories
- [x] Documentation updated (public API changes only) (PR: https://github.com/vectorgrp/sil-kit/pull/21)
- [ ] API docstrings updated (public API changes only)
- [ ] Rebase &rarr; commit history clean
- [ ] Squash and merge &rarr; proper PR title

### CODE REVIEW

**Review by KonradBkd — CHANGES_REQUESTED** [review]
Paar Kleinigkeiten + Changelog fehlt noch

File: SilKit/source/config/ParticipantConfigurationFromXImpl.cpp [hunk 1]
```diff
@@ -63,23 +105,475 @@ auto Parse(const std::string& text) -> SilKit::Config::ParticipantConfiguration
...
+    }
+}
+
+void CacheMiddleware(const YAML::Node& root, MiddlewareCache& cache)
+{
+    if (root["AcceptorUris"])
+    {
+        if (cache.acceptorUris.size() > 0)
+        {
+            throw SilKit::ConfigurationError{"AccepoorUris already defined!"};
```
[the remark this answers is not in the document]
Comment (VJanKraemer) [reply SilKit/source/config/ParticipantConfigurationFromXImpl.cpp]: Fixed :)

File: SilKit/source/config/ParticipantConfigurationFromXImpl.cpp [hunk 2]
```diff
@@ -63,23 +105,475 @@ auto Parse(const std::string& text) -> SilKit::Config::ParticipantConfiguration
...
+        for (const auto& include : config["Includes"]["Files"])
+        {
+            levelIncludes.push_back(include.as<std::string>());
+        }
+    }
+}
+
+// ================================================================================
+//  Helper functions to find and open config snippets
+//  TODO: Put into the Filesystem Layer?
```
[the remark this answers is not in the document]
Comment (VJanKraemer) [reply SilKit/source/config/ParticipantConfigurationFromXImpl.cpp]: Fixed :)

**Review by KonradBkd — APPROVED** [review]
LGTM!

**Pull request merged**